tests/src/main/java/net/neoforged/neoforge/debug/transfer/VanillaHandlersTests.java

```diff
@@ -0,0 +1,566 @@
+import net.minecraft.world.level.block.entity.AbstractFurnaceBlockEntity;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.BrewingStandBlockEntity;
+import net.minecraft.world.level.block.entity.ChestBlockEntity;
+import net.minecraft.world.level.block.entity.ChiseledBookShelfBlockEntity;
+import net.minecraft.world.level.block.entity.FurnaceBlockEntity;
+import net.minecraft.world.level.block.entity.HopperBlockEntity;
+import net.minecraft.world.level.block.entity.JukeboxBlockEntity;
+import net.minecraft.world.level.block.entity.ShulkerBoxBlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.gameevent.GameEvent;
+import net.neoforged.fml.util.ObfuscationReflectionHelper;
+import net.neoforged.neoforge.event.VanillaGameEvent;
+import net.neoforged.neoforge.transfer.handlers.wrappers.items.ComposterWrapper;
+import net.neoforged.neoforge.transfer.handlers.wrappers.items.LivingEntityEquipmentWrapper;
+import net.neoforged.neoforge.transfer.handlers.wrappers.items.PlayerInventoryWrapper;
+import net.neoforged.neoforge.transfer.handlers.wrappers.items.VanillaContainerWrapper;
+import net.neoforged.neoforge.transfer.handlers.wrappers.items.WorldlyContainerWrapper;
+import net.neoforged.neoforge.transfer.resources.ItemResource;
+import net.neoforged.neoforge.transfer.transaction.Transaction;
+import net.neoforged.testframework.DynamicTest;
+import net.neoforged.testframework.annotation.ForEachTest;
+import net.neoforged.testframework.annotation.TestHolder;
+import net.neoforged.testframework.gametest.EmptyTemplate;
+import net.neoforged.testframework.gametest.ExtendedGameTestHelper;
+import net.neoforged.testframework.gametest.GameTest;
+
+@ForEachTest(groups = "transfer.vanillahandlers")
+public class VanillaHandlersTests {
+    private static final ItemResource RESOURCE = ItemResource.of(Items.APPLE);
+
+    @GameTest
+    @EmptyTemplate
+    @TestHolder(description = "Test cooldown handling of the hopper wrapper")
+    public static void testHopperCooldown(ExtendedGameTestHelper helper) {
+        var pos = new BlockPos(1, 1, 1);
+        helper.setBlock(pos, Blocks.HOPPER);
+
+        // Note: Cooldown is initialized to -1, which also counts as not having cooldown.
+        var hopperEntity = helper.getBlockEntity(pos, HopperBlockEntity.class);
+        var hopper = VanillaContainerWrapper.of(hopperEntity);
+
+        // Insertion into empty hopper -> cooldown
+        try (var transaction = Transaction.open(null)) {
+            hopper.insert(RESOURCE, 10, transaction);
+            transaction.commit();
+        }
+        helper.assertValueEqual(HopperBlockEntity.MOVE_ITEM_SPEED, getHopperCooldown(hopperEntity), "hopper cooldown");
+        hopperEntity.setCooldown(0);
+
+        // Second insertion into hopper -> no cooldown
```

Comment:
Please clarify that the reason it is expected there won't be a cooldown, is because the hopper isn't empty.